tests/error_handling_test.go

```diff
@@ -0,0 +1,96 @@
+package tests
```

Comment:
Can we add a test for badly formatted requests.
E.g. try to run the relay and call `(POST)/userOperation` with the following body:
```json
{
    "ab": "cd"
}
```
Instead of returning `ErrMalformedJson` the goroutine crashes. We might need to add custom unmarshalling.

Reply:
done - add a method `IsZero` upon which a new error of unexpected JSON is passed